src/sentry/web/frontend/base.py

```diff
@@ -189,15 +189,33 @@ def get_active_project(self, request: Request, organization, project_slug):
     def redirect_to_org(self, request: Request):
         from sentry import features
 
+        using_customer_domain = request and is_using_customer_domain(request)
+
         # TODO(dcramer): deal with case when the user cannot create orgs
         if self.active_organization:
-            url = Organization.get_url(self.active_organization.organization.slug)
+            current_org_slug = self.active_organization.organization.slug
+            url = Organization.get_url(current_org_slug)
         elif not features.has("organizations:create"):
             return self.respond("sentry/no-organization-access.html", status=403)
         else:
+            org_exists = False
             url = "/organizations/new/"
-            if request.subdomain:
+            if using_customer_domain:
                 url = absolute_uri(url)
+
+            if using_customer_domain and request.user and request.user.is_authenticated:
+                organizations = organization_service.get_organizations(
+                    user_id=request.user.id, scope=None, only_visible=True
+                )
+                maybe_org = organization_service.get_organization_by_slug(
```

Comment:
If you don't need the entire organization object, try using `check_organization_by_slug` instead, which will only load the id or None.  These endpoints will otherwise be doing a lot of preloading to optimize for single RPC call -> several related objects.